src/serde.rs

```diff
@@ -826,4 +826,18 @@ mod test {
         let deserialized: StringExample = serde_json::from_str(r#"{"value":null}"#).unwrap();
         assert_eq!(deserialized.value, original.value);
     }
+
+    #[test]
+    fn with_overflow() {
+        #[derive(Serialize, Deserialize)]
+        pub struct OverflowExample {
+            value: Decimal,
+        }
+        // Going a fraction about the MAX number expecting it to fail but not panic.
+        let res: Result<Decimal, _> = Decimal::from_str(format!("{}.999999", Decimal::MAX).as_str());
```

Comment:
Instead of catching the expected deserialization error, it panicked like this:
```
        thread 'serde::test::with_overflow' panicked at 'assertion failed: `(left == right)`
        left: `1`,
        right: `0`', src/str.rs:384:5
```

I would prefer being able to throw any string at Decimal::str rather than implementing a separate validation layer.